File: AlKutub/app/src/main/java/com/example/al_kutub/model/AppNotification.kt
package com.example.al_kutub.model

import com.google.gson.annotations.SerializedName

data class AppNotification(
    @SerializedName("id")
    val id: Int,
    
    @SerializedName("title")
    val title: String,
    
    @SerializedName("message")
    val message: String,
    
    @SerializedName("type")
    val type: String,
    
    @SerializedName("action_url")
    val actionUrl: String? = null,
    
    @SerializedName("data")
    val data: String? = null,
    
    @SerializedName("read_at")
    val readAt: String? = null,
    
    @SerializedName("created_at")
    val createdAt: String,
    
    @SerializedName("updated_at")
    val updatedAt: String? = null
)

File: al-kutub/app/Models/AppNotification.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class AppNotification extends Model
{
    use HasFactory;

    protected $table = 'app_notifications';

    protected $fillable = [
        'title',
        'message',
        'type',      // 'info', 'new_kitab', 'promo', etc.
        'action_url', // URL optional (misal link ke detail kitab)
        'data',
    ];

    protected $casts = [
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    public function userReads()
    {
        return $this->hasMany(NotificationUserRead::class, 'notification_id');
    }
}
